float64[] k_array